src/main/java/org/folio/circulation/domain/Loan.java

```diff
@@ -57,10 +59,15 @@
 
   private final LoanPolicy loanPolicy;
 
+  private final OverdueFinePolicy overdueFinePolicy;
+  private final LostItemPolicy lostItemPolicy;
+
   private Loan(JsonObject representation, Item item, User user, User proxy,
-    ServicePoint checkinServicePoint, ServicePoint checkoutServicePoint,
-    DateTime originalDueDate, LoanPolicy loanPolicy,
-    Collection<Account> accounts) {
+               ServicePoint checkinServicePoint, ServicePoint checkoutServicePoint,
+               DateTime originalDueDate, LoanPolicy loanPolicy,
+               OverdueFinePolicy overdueFinePolicy,
```

Comment:
As there are already a lot of parameters when constructing a loan, I think it would be useful to extract all of the policies to a separate object that is a single parameter.

Does this suggestion make sense?

Reply:
The constructor has private modifier, and only the internal builder calls this constructor to provide an instance.

Reply:
How does the constructor being private affect the number of parameters being too high and triggering a sonar warning?

Reply:
Fixed